backend/routes/posts.js

```diff
@@ -0,0 +1,138 @@
+import express from "express";
+import { Post } from "../models/post.js";
+import { User } from "../models/user.js";
+import { PostComment } from "../models/postComment.js";
+
+const router = express.Router();
+
+router.get("/post/:id", async (req, res) => {
+    const { id } = req.params;
+
+    try {
+        const post = await Post.findOne({
+            where: { id },
+            include: [{ model: User }]
+        });
+
+        if (post === null) {
+            res.status(404).json({ error: "This Post Does Not Exist." });
+            return;
+        }
+
+        res.status(200).json({ post });
+    } catch (error) {
+        res.status(500).json({ error });
+    }
+});
+
+router.get("/post/comments/:id", async (req, res) => {
```

Comment:
It's the Post ID, not the comment ID. So, would it make more sense to have this as: `"/post/:id/comments"`